### DISCUSSION TITLE
Comparing map legibility for cycling

### DISCUSSION THREAD

**Discussion by fdr** [organicmaps/organicmaps#7071] [Ideas]
I had reviewed a cycle map of Reno, Nevada before a trip there, and found it appeared to be a bicycle-hostile city. But when I got there, I saw quite a few bicycle lanes; something wasn't adding up. I found Organic Maps while trying to figure out what map program I should use, and while I liked a lot about it, it wasn't faring so well in rendering Reno for bicycling.

I started comparing various map programs and found the legibility of this influential infrastructure was pretty poor on average, including the venerable OpenCycleMap.

Here are some comparisons.

OpenCycleMap HD, as seen at GaiaGPS. This is the reference I used before visiting Reno to determine it was a bicycle-hostile city, which is not correct. With a microscope, you might be able to pick out the blue bike paths on the side of roads, but this visually understates their importance in practical, safe, and pleasant use of a bicycle to accomplish tasks in Reno.

![image](https://github.com/organicmaps/organicmaps/assets/7191/7c2b2619-e771-4706-b1f0-c34b666101b6)

Organic Maps, which seems to show even less:

![image](https://github.com/organicmaps/organicmaps/assets/7191/45721ab1-278c-4b84-8465-23f1505c914a)

CyclOSM. It's not a perfectly legible map, but it makes it clear there's a lot more going on. This was screenshotted from my phone, the non-mobile web site is a lot more usable. Even so...it gives you a better sense of the structure of what's going on. It has a somewhat complex legend of things like surface quality and bike lane type, low speed limits, etcetera.

![image](https://github.com/organicmaps/organicmaps/assets/7191/a4bd60a4-1042-4941-a2f0-bfbab374f46c)

Bikemap. It has a bit less information about the specific kind of safe/green route it gives, and it seems like it classifies certain low-speed and/or low traffic streets as green (seen at the upper left), which could be a permissible interpretation. Nevertheless, it does pick out all the bicycle lanes.

![image](https://github.com/organicmaps/organicmaps/assets/7191/bcea110f-b1dc-4214-b93f-01276bf92daf)

Cyclemaps.org has some highly legible vector map tiles, though with less speed information, I think.

![image](https://github.com/organicmaps/organicmaps/assets/7191/0cb6276f-6ca4-4786-95de-69555de989ba)

I also tried out OSMAnd and Pocket Earth and found their maps in a basically illegible category, save that OSMAnd supports CyclOSM. Most renderings under-weight non-major bicycle routes, even ones with separated bicycle lanes. I think Bikemap and CyclOSM give the best sense of how usable a bicycle is in Reno, and by a large margin.

One quirk in CyclOSM is it use of round quantities of metric speed limits (20km/h, 30km/h) which stunts its rendering usefulness in the US quite a bit; 30km/h is close enough to 20mph that rounding would improve map legibility quite a bit.

I think all the maps could be improved in displaying bicycle-friendliness in terms of vehicle speed limits vs level of separation and protection of the bike lane. For example, bicycling in areas with a 20mph limit and only two car lanes (one in each direction) at most I find pretty humane even without a lane, but 30mph is quite fine with bollards (the exceptional case of a 20mph zone on a four-lane road tends to mean people speed, once again fine with a separated lane, not so much with a non-separated one) A lane without sound protection next to a 40+mph road is once again not so good.  I would make comment on protected bike lanes, except they're a rare species in the United States that I can't come up with a generalization.

Outside Reno, I also saw some "bicycle routes" (as in, they are official, there's signs) alongside 45-55mph roads, which are probably survivable if unpleasant with a bike lane and would appear pretty nerve-wracking to me when they have narrow shoulders, which they do sometimes. I don't think CyclOSM displays the level of nerve required quite right; it does not color very fast roads, only slow ones.

![image](https://github.com/organicmaps/organicmaps/assets/7191/178d8ec5-07aa-4b1a-98c9-4cfc92b05ca0)

Bikemap elects to treat the fast, bicycle-lane-free segment as essentially bicycle-hostile (it looks like any other freeway), which for most of us is a fair interpretation:
![image](https://github.com/organicmaps/organicmaps/assets/7191/5c406e2c-aad6-4017-9a90-0202e94f6f0d)

Cyclemaps.org takes the official route designation at its word, much as CyclOSM does. So, it's rather interesting that Bikemap is inclined to exclude the hairy no-bike lane small-shoulder region:

![image](https://github.com/organicmaps/organicmaps/assets/7191/a87fb9f0-dbec-4371-882e-bfc70eff1a0e)

While Bikemap represents something closer to the truth in this case for all but the most committed bicyclists, I think both approaches are less satisfactory than flagging fast roads, regardless of their bicycle infrastructure or official bicycle route status.

**Comment by pastk** [comment]
Thanks for your feedback!

There is a plan to introduce a separate map style / layer for cycling:
- #2360

**Comment by patepelo** [comment]
@fdr What do you think of the latest changes made to improve this?

**Reply by fdr** [reply]
Huh, I just installed a new version from testflight, and I still don't see a bicycle map? Only {outdoor, contour, subway} like before. None of them show the lanes where I live, but the data is definitely present (San Francisco)

**Reply by patepelo** [reply]
> Huh, I just installed a new version from testflight, and I still don't see a bicycle map? Only {outdoor, contour, subway} like before. None of them show the lanes where I live, but the data is definitely present (San Francisco)

The main map is now more prominent in regards to bicycle lanes being blue. A cycling layer is still pending and planned. If you're interested I may ping/cc you on related issues.

**Reply by pastk** [reply]
@fdr in the latest version there should be a better / more consistent depiction of segregated cycleways vs shared ones, for example.
But maybe it doesn't make much difference to your region, esp. if its mostly on-road bike lanes (which are not displayed in OM yet).

**Reply by manouche** [reply]
Yes, for me it's still unreadable 🥲 sorry. ![image](https://github.com/organicmaps/organicmaps/assets/131271/9652eeb9-9836-49b3-947d-d44b7f1dde89)

**Reply by fdr** [reply]
> @fdr in the latest version there should be a better / more consistent depiction of segregated cycleways vs shared ones, for example.
> 
> But maybe it doesn't make much difference to your region, esp. if its mostly on-road bike lanes (which are not displayed in OM yet).

Hmm that's not what does for sure. Laguna Honda blvd is definitely separated, though it has no concrete barrier. Here's what it looks like. ![image](https://github.com/organicmaps/organicmaps/assets/7191/a99f4705-a096-4b64-b947-b2e782c11ecb)

**Reply by pastk** [reply]
ATM its gotta to be mapped as a separate way (e.g. `highway=cycleway`) to be displayed in OM. Any kind of on-road bicycle infrastructure tags are not displayed.
It'll be implemented in a separate bicycle map style.

**Comment by michaelblyons** [comment]
Were you using the cycle map style for OsmAnd?

![Screenshot_20240604-155833](https://github.com/organicmaps/organicmaps/assets/2701605/301c42fe-e2f5-42d2-8781-3614bce094da)

![Screenshot_20240604-191722](https://github.com/organicmaps/organicmaps/assets/2701605/6c49edb0-6235-47cc-b6f0-85896a4d3d2a)

![Screenshot_20240604-192413](https://github.com/organicmaps/organicmaps/assets/2701605/dfab3dfc-1f4c-4680-bf45-20ab45b85ea9)

**Reply by Coehill** [reply]
For posterity, OSMAnd~ uses CyclOSM for its cycle map style.

**Reply by michaelblyons** [reply]
Wait, what? I know I can add CyclOSM as an online tile provider. Is there an offline style, too?

**Reply by Coehill** [reply]
Not sure if it's online or not but you can use CyclOSM from the _Map Source_ setting in ~OSMAnd.

![OSMAnd_screenshot](https://github.com/user-attachments/assets/67ef90b3-4e8a-4a34-8a90-4cef27bb948d)

**Reply by michaelblyons** [reply]
Ah, okay. We're taking about different things, then. My screenshots are from the offline vector maps, and then further down I set map options to use the Touring palette and overlay bike routes.

**Comment by Coehill** [comment]
> One quirk in CyclOSM is it use of round quantities of metric speed limits (20km/h, 30km/h) which stunts its rendering usefulness in the US quite a bit; 30km/h is close enough to 20mph that rounding would improve map legibility quite a bit.

Please consider opening a ticket is [cyclosm's issue tracker](https://github.com/cyclosm/cyclosm-cartocss-style/issues) so this can be resolved.

**Comment by Coehill** [comment]
> Outside Reno, I also saw some "bicycle routes" (as in, they are official, there's signs) alongside 45-55mph roads, which are probably survivable if unpleasant with a bike lane and would appear pretty nerve-wracking to me when they have narrow shoulders, which they do sometimes. I don't think CyclOSM displays the level of nerve required quite right; it does not color very fast roads, only slow ones.
> 
> Bikemap elects to treat the fast, bicycle-lane-free segment as essentially bicycle-hostile (it looks like any other freeway), which for most of us is a fair interpretation:
>
> Cyclemaps.org takes the official route designation at its word, much as CyclOSM does. So, it's rather interesting that Bikemap is inclined to exclude the hairy no-bike lane small-shoulder region:
>
> While Bikemap represents something closer to the truth in this case for all but the most committed bicyclists, I think both approaches are less satisfactory than flagging fast roads, regardless of their bicycle infrastructure or official bicycle route status.

Tl;dr: This appears to be a data issue.

Looking at how Mount Rose Highway is marked as a local bike route, despite having a speed limit of 45mph (72km/h), something seemed off.  It's marked as a bike route for a very long stretch of road.  From looking at street-level imagery, I saw a dangerous, twisty road with no shoulder.  From a few spot-checks, as well as searching Mapillary for auto-tagged signage, I did not see any bicycle route signage.

I did see that there is a shorter section of adjacent roadway called Old Mount Rose Highway which is showing up on many local bicycling routes and appears to be a safe bike route.  Unfortunately, there is no Panoramax, Mapillary, or Google Street View imagery for this road.

My theory is that a mapper accidentally marked Mount Rose Highway as a lcl=yes when they should have mapped Old Mount Rose Highway as lcn=yes.  @fdr Since you know the area, can you please confirm if this may be true? :)

IMHO it makes sense to trust that an official signed bike route IS safe to ride on and display it like CyclOSM is without prioritizing the speed limit if no other data is available.  CyclOSM/OpenStreetMap did not have any shoulder or bike lane information (see StreetComplete screenshot below) so it's a reasonable assumption based on the data it had available.

Consider this scenario: 
If all OSM knows, just like before, is that the speed limit is 45mph and it's a signed bike route, but no data on shoulders/bicycle infrastructure.  Then there could be a separate bicycle path but we don't know because we lack the data.  So if we prioritize the speed limit, then the road would be marked as unsafe when it's really the safest route.  Since it's more likely than not that a signed bike route was placed in a safe area, then we should trust it over the speed limit UNLESS we have complete data of shoulder, surface type, bicycle infrastructure or lack thereof, etc, to prove that it's an edge case.

If there is a signed bike route on an unsafe road, then that points to either underlying data issues, or a highly unlikely edge case where signage was placed on unsafe roads.

It would be awesome if someone could photo-map Old Mount Rose Highway with Panoramax and add any available speed limit / bike route signage to both Old Mount Rose Highway and Mount Rose Highway!  Also, break it up where there are/aren't shoulders, etc.

![Zoomed out ID editor screenshot of Mount Rose Highway being one single roadway](https://github.com/user-attachments/assets/395e6692-ac7a-460d-a052-4b5c8f24a972)
![Screenshot of Mount Rose Highway tags](https://github.com/user-attachments/assets/31f09ee7-bd6f-4d96-b097-aebd15439cf9)
StreetComplete:
![Screenshot of StreetComplete showing the lack of bicycle infrastructure tagging of Mount Rose Highway](https://github.com/user-attachments/assets/a2c106bd-ba7e-4a30-a0e6-4acf662058f5)
